### DISCUSSION TITLE
revoke column level security

### DISCUSSION THREAD

**Discussion by roo12312** [supabase/supabase#20791] [Questions]
Hi  
`revoke insert(id) on public.demo from authenticated;`
running this SQL statement it returns `Success. No rows returned` 
but when I try to insert with id it succeeds.

using the column security preview feature when I turn on and off the id insert field and then run the same SQL command it starts to restrict.

am i doing something wrong or is this is a bug?

**Comment by GaryAustin1** [comment]
You should not report a bug if you are not sure or close to sure.  Discussion in github supabase or Discord are better starting points.
Not sure what the preview feature has to do with your executing SQL.
You can't just revoke a column grant on a table and have it work.  First you have to remove table grant for the operation from the role.  Then you have to ADD the column grants you want.  Then you can remove one with revoke if you like.

Sorry if I misunderstand your issue.

**Comment by roo12312** [comment]
hi i followed this guide `https://supabase.com/docs/guides/auth/column-level-security`  and run manual sql `revoke insert(id) on public.demo from authenticated;` but it doesn't work.

Also 

> First you have to remove table grant for the operation from the role. Then you have to ADD the column grants you want. Then you can remove one with revoke if you like.

revoke is the process to remove the table grant access right?

**Comment by roo12312** [comment]
this is a simple reproduction 

```

create table if not exists
public.posts (
id bigint primary key generated always as identity,
user_id uuid not null default auth.uid (),
title text,
content text
);

-- Add row-level security

alter table public.posts enable row level security;


create policy  "Allow update for owners" on public.posts for
update
using (auth.uid () = user_id);



-- Add column-level security
revoke
update
(title) on table public.posts
from
authenticated;

```

**Comment by GaryAustin1** [comment]
That is pure SQL and does not involve Supabase.
You did not revoke the role from doing the table update operation.  Table operations take priority over column privileges.  By default all the roles have table grants for all operations.  You have to revoke the table update grant before you can do column stuff.

`A user may perform SELECT, INSERT, etc. on a column if he holds that privilege for either the specific column or its whole table. Granting the privilege at the table level and then revoking it for one column will not do what you might wish: the table-level grant is unaffected by a column-level operation.`  from https://www.postgresql.org/docs/9.1/sql-grant.html

I'm moving this to a discussion for now as does not seem to be a Supabase bug.
